crates/sui-types/src/messages.rs

```diff
@@ -787,7 +787,10 @@ impl TransactionData {
     }
 
     pub fn input_objects(&self) -> SuiResult<Vec<InputObjectKind>> {
-        let mut inputs = self.kind.input_objects()?;
+        let mut inputs = self
+            .kind
+            .input_objects()
+            .map_err(|e| SuiError::TransactionInputObjectsErrors { errors: vec![e] })?;
```

Comment:
The wrapping operation seems super common.  Should we consider adding:

```
impl SuiError {
    pub fn transaction_input(error: SuiError) -> SuiError {
        SuiError::TransactionInputObjectsErrors { errors: vec![error] }
    }
}
```

That way this and other call-sites become:

```suggestion
            .map_err(SuiError::transaction_input)?;
```

Reply:
good idea!